### PR TITLE
QE: Adjusting small API errors

### PR DESCRIPTION
**Pull request by elariekerboull**
## What does this PR change?

Makes small adjustments to API calls while following the API doc.

- `activationkey.rb` correcting parameters values
- ~~`api.rb` adjusting `get_version` and `system_version` calls so they include a second parameter that is required (empty Hash)~~
- ~~`http_client.rb` to include `system_version` as a GET request and accept a default empty argument on api call~~
- ~~`xmlrpc_client.rb` to accept a default empty argument on api call~~
- Remove paraemeter-less api calls in `api.rb`
## GUI diff

No difference.

Before:

After:

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links
Ports :
4.2
4.3

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by elariekerboull** [comment]
Just realised that instead of sending empty Hashes in `api.rb`, I could change the called functions to accept an empty Hash as default parameter, not sure what would be best.

**Review by srbarrios — APPROVED** [review]

**Review by vandabarata — APPROVED** [review]

**Review by nodeg — APPROVED** [review]
LGTM but I am relying on your PR test run :wink:

File: testsuite/features/support/http_client.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def initialize(host)
   def prepare_call(name, params)
     short_name = name.split('.')[-1]
     call_type =
-      if short_name.start_with?('list', 'get', 'is', 'find') ||
+      if short_name.start_with?('list', 'get', 'is', 'find', 'system') ||
```
Comment (Bischoff) [inline testsuite/features/support/http_client.rb:16]: shouldn't you then remove the test for `system.search` just below?

or is your new test duplicate and therefore useless?

File: testsuite/features/support/namespaces/activationkey.rb [hunk 2]
```diff
@@ -39,11 +39,11 @@ def verify(id)
   end
 
   def add_config_channels(id, config_channels)
-    @test.call('activationkey.addConfigChannels', sessionKey: @test.token, keys: id, configChannels: config_channels, addToTop: false)
+    @test.call('activationkey.addConfigChannels', sessionKey: @test.token, keys: id, configChannelLabels: config_channels, addToTop: false)
```
Comment (Bischoff) [inline testsuite/features/support/namespaces/activationkey.rb:42]: thank you

File: testsuite/features/support/http_client.rb [hunk 3]
```diff
@@ -33,9 +33,10 @@ def prepare_call(name, params)
     [call_type, url]
   end
 
-  def call(name, params)
+  def call(name, *parameters)
+    params = parameters.first
     # Get session cookie from previous calls
-    if params.nil?
+    if params.nil? || params.empty?
```
Comment (Bischoff) [inline testsuite/features/support/http_client.rb:39]: I do not understand. If `parameters` is empty, won't `params` be `nil` ?

File: testsuite/features/support/http_client.rb [hunk 4]
```diff
@@ -33,9 +33,10 @@ def prepare_call(name, params)
     [call_type, url]
   end
 
-  def call(name, params)
+  def call(name, *parameters)
+    params = parameters.first
```
Comment (Bischoff) [inline testsuite/features/support/http_client.rb:37]: ouch. no, this seems utterly wrong, you need to pass all parameters

File: testsuite/features/support/xmlrpc_client.rb [hunk 5]
```diff
@@ -10,7 +10,8 @@ def initialize(host)
     @xmlrpc_client = XMLRPC::Client.new2('https://' + host + '/rpc/api', nil, DEFAULT_TIMEOUT)
   end
 
-  def call(name, params)
+  def call(name, *parameters)
+    params = parameters.first || {}
```
Comment (Bischoff) [inline testsuite/features/support/xmlrpc_client.rb:14]: what motivates this ?

Comment (Bischoff) [inline testsuite/features/support/xmlrpc_client.rb:14]: same remark, no first

**Comment by elariekerboull** [comment]
After discussing with Eric, it seemed simpler to remove api calls that did not include parameters, especially since they are never used.

**Review by Bischoff — APPROVED** [review]
thanks for the changes

"do not forget the ports" :-)

**Pull request merged**